Classes/Emogrifier.php

```diff
@@ -190,25 +210,32 @@ public function emogrify()
         if ($nodesWithStyleAttributes !== false) {
             /** @var \DOMElement $node */
             foreach ($nodesWithStyleAttributes as $node) {
-                $normalizedOriginalStyle = preg_replace_callback(
-                    '/[A-z\\-]+(?=\\:)/S',
-                    function (array $m) {
-                        return strtolower($m[0]);
-                    },
-                    $node->getAttribute('style')
-                );
-
-                // in order to not overwrite existing style attributes in the HTML, we have to save
-                // the original HTML styles
-                $nodePath = $node->getNodePath();
-                if (!isset($this->styleAttributesForNodes[$nodePath])) {
-                    $this->styleAttributesForNodes[$nodePath] = $this->parseCssDeclarationBlock(
-                        $normalizedOriginalStyle
+                // If inline styles parsing is enabled, normalize the "style" attribute and save it
+                if ($this->isInlineStylesParsingEnabled) {
+                    $normalizedOriginalStyle = preg_replace_callback(
```

Comment:
While you're changing is anyway, could you please move the code block within the if to a separate private method?